### PR TITLE
docs: Example of SARIF adapter implementation

### PR DESCRIPTION
**Pull request by petertrr**

### CODE REVIEW

File: example/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+plugins {
```
Comment (orchestr7) [inline example/build.gradle.kts]: need extra CSV example

File: example/src/main/kotlin/org/cqfn/save/example/ExampleAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import kotlinx.serialization.json.Json
+import org.cqfn.save.adapter.Adapter
+import kotlin.reflect.typeOf
+
+class ExampleAdapter : Adapter<List<Example>>(
+    type = typeOf<List<Example>>(),
+    decoder = Json,
+    tool = Tool(driver = ToolComponent(name = "my-awesome-tool")),
+) {
+    override fun mapToSarifResults(t: List<Example>): List<Result> {
```
Comment (orchestr7) [inline example/src/main/kotlin/org/cqfn/save/example/ExampleAdapter.kt]: let's make it as lambda

File: example/src/main/kotlin/org/cqfn/save/example/ExampleAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import io.github.detekt.sarif4k.Tool
+import io.github.detekt.sarif4k.ToolComponent
+import kotlinx.serialization.json.Json
+import org.cqfn.save.adapter.Adapter
+import kotlin.reflect.typeOf
+
+class ExampleAdapter : Adapter<List<Example>>(
+    type = typeOf<List<Example>>(),
+    decoder = Json,
+    tool = Tool(driver = ToolComponent(name = "my-awesome-tool")),
```
Comment (orchestr7) [inline example/src/main/kotlin/org/cqfn/save/example/ExampleAdapter.kt]: let's hardcode a default value

File: example/src/main/resources/META-INF/services/org.cqfn.save.adapter.AdapterProxy [hunk 4]
```diff
@@ -0,0 +1 @@
+org.cqfn.save.example.ExampleAdapterProxy
```
Comment (orchestr7) [inline example/src/main/resources/META-INF/services/org.cqfn.save.adapter.AdapterProxy]: JsonExample

**Review by orchestr7 — APPROVED** [review]

File: example/src/main/kotlin/org/cqfn/save/example/ExampleAdapter.kt (hunk 2 above)
Comment (petertrr) [reply example/src/main/kotlin/org/cqfn/save/example/ExampleAdapter.kt]: Done after refactoring of `Adapter`

File: example/src/main/kotlin/org/cqfn/save/example/ExampleAdapter.kt (hunk 3 above)
Comment (petertrr) [reply example/src/main/kotlin/org/cqfn/save/example/ExampleAdapter.kt]: Done

File: example/src/main/resources/META-INF/services/org.cqfn.save.adapter.AdapterProxy (hunk 4 above)
Comment (petertrr) [reply example/src/main/resources/META-INF/services/org.cqfn.save.adapter.AdapterProxy]: Renamed both class and directory

**Pull request merged**